ftprime/argrecorder.py

```diff
@@ -304,20 +370,61 @@ def simplify(self, samples):
         self.update_times()
         sample_nodes = self.get_nodes(samples)
         if self.timings is not None:
-            start = timer.process_time()
-        msprime.sort_tables(nodes=self.nodes, edges=self.edges,
-                            sites=self.sites, mutations=self.mutations)
-        #                   migrations=self.migrations)
+            before = timer.process_time()
+
+        if self.last_sorted_edge < self.edges.num_rows:
+            # begin modified block from @jeromekelleher and @molpopgen
+            # Copy the already sorted edges to local arrays
+            left = self.edges.left[:self.last_sorted_edge]
+            right = self.edges.right[:self.last_sorted_edge]
+            parent = self.edges.parent[:self.last_sorted_edge]
+            child = self.edges.child[:self.last_sorted_edge]
+            # Get the new edges and reverse them. After this, we know that all edges
+            # are correctly sorted with respect to time. We then sort each time slice
+            # individually, reducing the overall cost of the sort.
```

Comment:
OK doh. I've realized that this clearly isn't true with overlapping generations

Reply:
Doh x2. Overlapping generations means that chunk-wise simplify just won't work, at least not without some cleverer bookkeeping. Let's say we have *n* existing edges sorted by parent birth time for a node table containing times up to *t*. Then assume we are adding edges at later, say _t+2_.

If we want to not have to sort *k* < *n* edges, then we can only take edges up till we reach an edge whose parent is living.